pkg/applicationserver/io/packages/loradms/v1/package.go

```diff
@@ -74,6 +74,12 @@ func (p *DeviceManagementPackage) HandleUp(ctx context.Context, def *ttnpb.Appli
 		return nil
 	}
 
+	defer func() {
+		if err != nil {
+			events.Publish(evtPackageFail.NewWithIdentifiersAndData(ctx, up.EndDeviceIdentifiers, err))
```

Comment:
Prefer using the `registerXXX()` pattern so that we can implement metrics in one place (`observability.go`)